modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationModal.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayManagementToolbar from '@clayui/management-toolbar';
+import ClayModal from '@clayui/modal';
+import ClayTable from '@clayui/table';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const TranslationModal = ({
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	defaultLocale,
+	locales,
+	observer,
+	translations,
+}) => {
+	const [active, setActive] = React.useState(false);
+	const [icon, setIcon] = React.useState('search');
+	const inputRef = React.createRef();
+	const [visibleLocales, setVisibleLocales] = React.useState(locales);
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayModal.Header>
+				{Liferay.Language.get('manage-translations')}
+			</ClayModal.Header>
+
+			<ClayModal.Body scrollable>
+				<ClayManagementToolbar aria-label="Management toolbar">
+					<ClayManagementToolbar.Search showMobile={true}>
+						<ClayInput.Group>
+							<ClayInput.GroupItem>
+								<ClayInput
+									aria-label={Liferay.Language.get('search')}
+									className="form-control input-group-inset input-group-inset-after"
+									onInput={() => {
+										const value = inputRef.current?.value;
+
+										const icon = value ? 'times' : 'search';
+										setIcon(icon);
+
+										if (!value) {
+											setVisibleLocales(locales);
+
+											return;
+										}
+
+										const result = Object.values(
+											visibleLocales
+										).filter((locale) => {
+											return locale.label.match(
+												new RegExp(value, 'i')
+											);
+										});
```

Comment:
There are some complicated cases of dealing with filters this way, when you use the same information source and update it filtered, as this is a controlled filter with a limited list, the best world for this is to keep the filter being done in time rendering anyway this is being done too but you can handle the filter being done inside a `useMemo` with a dependency to a `value`, this is quite common to see in some like this and a little more reactive to changing the value.

For example, the value of `visibleLocales` will be reset to the original value which is `locales` only if the user clicks on the icon, this usually doesn't cover all interactions, the user can just clear all entered values and the behavior default of listing all values are not listed, as you type the filter it will be based on the most recent `visibleLocales`, so if the user gives up on what he just typed and then writes a new value, the chance of not showing the correct result is very high, in this case, the original state should always be considered instead of the recent state of the filter.

Controlling all these cases makes it difficult when you need to filter and store a state, so I have a preference and much more common in other applications and cases like autocomplete are reactive to `value` at render time.

```jsx
const filteredLocales = useMemo(() => {
  return Object.values(
    locales
  ).filter((locale) => locale.label.match(
    new RegExp(value, 'i')
  ))
}, [locales, value]);
```

I'm not sure but `locales` already looks like an `Array` so it doesn't need an `Object.values(locales)`.

Reply:
Thanks for the explanation @matuzalemsteles. I'll use the `useMemo` hook, and check if  `Object.values` is really needed here

Reply:
Updated in [`8a9dfad`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/8a9dfad58eeeec19dfd11eed9e325170680b3266) - sorry this might be the "hardest" thing to review, but it does look nicer